src/StrategyReserves.sol

```diff
@@ -0,0 +1,211 @@
+ */
+contract StrategyReserves is Ownable {
+    using SafeERC20 for IERC20;
+
+    /// @notice The ERC20 token contract
+    IERC20 public immutable token;
+
+    /// @notice Array of available strategies
+    address[] public strategies;
+
+    /// @notice Mapping to track which strategies have been added for O(1) duplicate checking
+    mapping(address => bool) public strategyExists;
+
+    /// @notice Minimum amount required to claim a strategy (in token's native decimals)
+    uint256 public minimumClaimAmount = 1e6; // Default: 1 token (assuming 6 decimals)
+
+    /// @notice Emitted when a strategy is added to the contract
+    event StrategyAdded(address indexed strategy);
+
+    /// @notice Emitted when a strategy is claimed by a user
+    event StrategyClaimed(address indexed user, address indexed strategy, uint256 amount);
+
+    /// @notice Emitted when a strategy is claimed on behalf of an address
+    event StrategyClaimedForAddress(address indexed beneficiary, address indexed strategy);
+
+    /// @notice Emitted when the minimum claim amount is updated
+    event MinimumClaimAmountUpdated(uint256 oldAmount, uint256 newAmount);
+
+    /**
+     * @notice Constructor that initializes the factory with ERC20 token address
+     * @param _token Address of the ERC20 token
+     * @param _owner Address of the factory owner
+     */
+    constructor(address _token, address _owner, uint256 _minimumClaimAmount) Ownable(_owner) {
+        require(_token != address(0), "Invalid token address");
+        token = IERC20(_token);
+        minimumClaimAmount = _minimumClaimAmount;
+    }
+
+    /**
+     * @notice Adds a strategy to the available pool
+     * @dev The factory must be the owner of the strategy when adding. Duplicate strategies are not allowed.
+     * @param strategy The address of the strategy to add
+     */
+    function add(address strategy) external onlyOwner {
+        require(strategy != address(0), "Invalid strategy address");
+
+        IStrategyWithOwnership strategyContract = IStrategyWithOwnership(strategy);
+
+        // Verify that this contract is the owner of the strategy
+        require(strategyContract.owner() == address(this), "Factory must be the owner of the strategy");
+
+        // Check for duplicate strategy using O(1) mapping lookup
+        require(!strategyExists[strategy], "Strategy already exists");
+
+        strategies.push(strategy);
+        strategyExists[strategy] = true;
+
+        emit StrategyAdded(strategy);
+    }
+
+    /**
+     * @notice Claims a strategy from the pool by depositing tokens
+     * @dev Pops the last strategy from the list, transfers tokens from caller to contract,
+     *      transfers ownership to caller, deposits tokens, and verifies ownership transfer
+     * @param amount The amount of tokens to deposit (in token's native decimals)
+     */
+    function claim(uint256 amount) external {
```

Comment:
I like the idea of claim and deposit. This makes a DOOs attack more difficult, but I'm still concerned that we could soffer one because they can deposit and withdraw atomically. Wdyt @hishboy ?